src/Wallabag/CoreBundle/Repository/EntryRepository.php

```diff
@@ -86,6 +86,36 @@ public function getBuilderForStarredByUser($userId)
     }
 
     /**
+     * Retrieves entries filtered with a search term for a user.
+     *
+     * @param int    $userId
+     * @param string $term
+     * @param strint $currentRoute
+     *
+     * @return QueryBuilder
+     */
+    public function getBuilderForSearchByUser($userId, $term, $currentRoute)
+    {
+        $qb = $this
+            ->getBuilderByUser($userId);
+
+        if ('starred' === $currentRoute) {
+            $qb->andWhere('e.isStarred = true');
+        } elseif ('unread' === $currentRoute) {
+            $qb->andWhere('e.isArchived = false');
+        } elseif ('archive' === $currentRoute) {
+            $qb->andWhere('e.isArchived = true');
+        }
+
+        $qb
+            ->andWhere('e.content LIKE :term OR e.title LIKE :term')->setParameter('term', '%'.$term.'%')
```

Comment:
As you have Doctrine Extension already installed on the project, you could use https://github.com/beberlei/DoctrineExtensions/blob/master/src/Query/Mysql/MatchAgainst.php to do "real" token based full text search directly with MySQL 😋 
It has the advantage of allowing to sort the result by **relevance**.

```
$qb->addSelect('match_against(p.titleFR, p.bodyFR) against (:search boolean ) AS _score');
$qb->having('_score > 0');
$qb->addOrderBy('_score', 'DESC');
$qb->setParameter('search', implode(',', $searchTokens));
```

I personally use this method to compute search tokens from user search:

```
    private function getSearchTokens($search)
    {
        $search = str_replace('  ', '', $search);
        $search = str_replace(' ', ',', $search);
        $search = str_replace('-', ',', $search);
        $search = str_replace('+', '', $search);
        $search = Transliterator::urlize($search, ',');

        $tokens = array_filter(explode(',', $search));

        // Add "mandatory" prefix
        array_walk($tokens, function(&$item) { $item = '+'.$item; });

        return $tokens;
    }
```